script.js

```diff
@@ -0,0 +1,194 @@
+let input = document.getElementById('input');
+let reposContainer = document.getElementById('repos__container');
+let sortDetails = document.getElementById('sort__details');
+let curInputVal = '';
+let repoList = [];
+let displayedRepoList = [];
+let repoIds = new Set();
+let curSortCriteria = 'stars';
+
+function setSortCriteria(criteria) {
+    let sortBtns = document.querySelectorAll('.sort__btn');
+    for(let i=0; i<sortBtns.length; i++) {
+        sortBtns[i].style.backgroundColor = 'var(--light-grey)';
+    }
+    if(criteria == 'stars') {
+        sortBtns[0].style.backgroundColor = 'var(--light-blue)';
+    }
+    else if(criteria == 'popular') {
+        sortBtns[1].style.backgroundColor = 'var(--light-blue)';
+    }
+    else {
+        sortBtns[2].style.backgroundColor = 'var(--light-blue)';
+    }
+    if(criteria != curSortCriteria) {
+        curSortCriteria = criteria;
+        let sortInfo = document.createElement('b');
+        sortInfo.innerText = curSortCriteria[0].toUpperCase() + curSortCriteria.slice(1);
+    
+        sortDetails.innerHTML = 'Current sort criteria: ';
+        sortDetails.append(sortInfo);
+        
+        reposContainer.innerHTML = '';
+        showRepos();
+    }
+}
+
+/* Process input data on Enter */
+input.addEventListener('keypress', function(event) {
+    if(event.key == 'Enter'){
+        getInputVal();
+    }
+});
+
+function getInputVal() {
+    if(input.value != '' && input.value != curInputVal){
+        console.log(input.value);
+        repoList = [];
+        repoIds.clear();
+        curInputVal = input.value;
+        fetchAPI(input.value);
+    }
+}
+
+async function fetchAPI(name) {
+    let found = false;
+    let li = name.split(' ').join('');
+    const nameList = li.split(',');
+    reposContainer.innerText = '';
+    for(let i=0; i<nameList.length; i++){
```

Comment:
I would like the fetch function to be written to handle a single input rather than having the split and loop inside it. e.g:

```jsx
async function fetchUserRepos(userId){
....
}

function handleRender(repos){
...
}

async function handleInput(input){
   const list = input.split(',');
   list.forEach(function(userId){
      const repos = await fetchUserRepos(userId);
      handleRender(repos)
   })
}
```